app/services/update_team_name_and_ldu_service.rb

```diff
@@ -0,0 +1,42 @@
+class UpdateTeamNameAndLduService
+  def self.update(team_code:, team_name:, ldu_code:)
+    new(team_code: team_code, team_name: team_name, ldu_code: ldu_code).update
+  end
+
+  def initialize(team_code:, team_name:, ldu_code:)
+    @team = Team.find_by(code: team_code)
+    @team_name = team_name
+    @ldu_code = ldu_code
+
+    Rails.logger.error("Couldn't find team with code #{team_code}") if @team.nil?
+  end
+
+  def update
+    return if @team.nil?
+
+    update_name if name_needs_updating?
+    update_ldu if ldu_needs_updating?
+    @team.save if @team.changed?
+  end
+
+private
+
+  def name_needs_updating?
+    @team.name != @team_name
+  end
+
+  def update_name
+    @team.name = @team_name
+  end
+
+  def ldu_needs_updating?
+    @team.local_divisional_unit&.code != @ldu_code
+  end
+
+  def update_ldu
+    ldu = LocalDivisionalUnit.find_by(code: @ldu_code)
```

Comment:
What do you think about pulling this into the initializer? I think it might be clearer to have the "if any of this data looks suspect, just don't do anything" errors into one place.

I think that may also make the updating code clearer in turn, but I'd need to play with it a little.

Reply:
That's a fair point – I'd like the explicitness of having those 'find' errors in the same place.

The reasoning for this approach was to avoid performing (potentially) unnecessary database queries. In practice, I expect most teams will already have the correct LDU (that's the ideal scenario 😬🤞) so won't need updating – my intention was to avoid looking up the 'target' LDU until it's clear that we need to update the existing association.

Perhaps that's an over-optimisation at this stage 🤷 but I'd prefer to leave it as-is for now, so long as you're happy for me to do so.